### ISSUE TITLE
DMOP vs MOP (and preDMOP)

### ISSUE THREAD

**Issue by ionides** [pypomp/tutorials#2]
A source of confusion among referees for the [MOP paper](https://arxiv.org/abs/2407.03085) was that the algorithm for the automatic derivative of MOP at theta=phi does not include phi, but has a stop_gradient instead. Both referees struggled (in different ways) to see the link between the MOP pseudocode and the numerical implementation.

In the revision of this paper, we are calling the autodiff derivative estimation algorithm DMOP so we can distinguish it from MOP. MOP is basically a theoretical construct to help explain and analyze DMOP, but is seldom actually implemented (indeed, it is numerically unstable at theta not equal to phi).

We don't yet have a name for the algorithm which, when autodiff is applied to it, matches the gradient of MOP, i.e., the pre-derivative of DMOP. One could use MOP itself as the pre-derivative, but it is simpler and better to use the stop_gradient representation. Let's call this pre-derivative preDMOP.

preDMOP is a curious algorithm because it is a recipe for what grad should do. Usually, an algorithm has defined input and output; here, the output is only defined after you nest preDMOP within `jax.grad` (or any equivalent implementation of stop_gradient). preDMOP can be evaluated, in which case it matches `pfilter`, or you can apply `jax.grad` to get DMOP. I don't know if there's a computer science name for an algorithm that is only defined when operated on by another algorithm? 

People usually just think of autodiff as a way to differentiate an algorithm, so this issue doesn't arise. More properly, grad is a procedure that outputs a quantity that may or may not be a good estimator of a derivative, and so things get delicate. This distinction, which is usually not made explicit, becomes important when the ability of grad to provide a good estimate of a derivative is the question at stake! One MOP referee was worried about the lack of formal mathematical foundations for stop_gradient, which isn't really a lack so much as suppressed details. This suggests that we should aim to be somewhat more explicit, trying to combine mathematical formality with readability to add to clarity for those who want to think hard while also helping to build a simpler intuitive understanding.

One advantage of being careful about the relationships between MOP, preDMOP and DMOP is that it can help to emphasize that there is a strong mathematical foundation to these algorithms. Once they have distinct names, it is easier to talk about how they relate to each other.

I'm curious whether @hetankevin sees a role for preDMOP terminology in the MOP revision. Either way, there are synergies between explaining the software and explaining the theory.
